### PR TITLE
Content-Security-Policy compatibility

### PR DESCRIPTION
**Pull request by daniel-beck**
https://wiki.jenkins-ci.org/display/SECURITY/Jenkins+Security+Advisory+2015-12-09
https://wiki.jenkins-ci.org/display/JENKINS/Configuring+Content+Security+Policy

### CODE REVIEW

File: src/main/resources/htmlpublisher/HtmlPublisherTarget/HTMLAction/sidepanel.jelly [hunk 1]
```diff
@@ -1,8 +1,6 @@
 <!--
   This view is used to render the plugin side panel.
 
-  We don't really have anything here, This is only a placeholder. 
+  We don't really have anything here, This is only a placeholder.
 -->
-<div>
```
Comment (daniel-beck) [inline src/main/resources/htmlpublisher/HtmlPublisherTarget/HTMLAction/sidepanel.jelly]: When browsing the archived files directly, this `div` showed up repeatedly as it's not wrapped in a layout element.

File: pom.xml [hunk 2]
```diff
@@ -57,6 +57,7 @@
               <xmlOutput>true</xmlOutput>
               <findbugsXmlWithMessages>true</findbugsXmlWithMessages>
               <failOnError>${findbugs.failOnError}</failOnError>
+              <omitVisitors>UnreadFields</omitVisitors>
```
Comment (daniel-beck) [inline pom.xml]: I failed to have FindBugs suppress a false positive for a field used in the Groovy view only, so used this workaround.

**Comment by daniel-beck** [comment]
Someone filed [JENKINS-32026](https://issues.jenkins-ci.org/browse/JENKINS-32026) for the issue this PR resolves.

@mrooney I'd prefer if you could release this fix.

**Comment by mrooney** [comment]
Thanks, released as 1.10!

**Comment by daniel-beck** [comment]
@mrooney Thanks! I'll update the wiki page.

**Comment by dubrsl** [comment]
It helped partially.
CSS not shown

**Comment by daniel-beck** [comment]
@dubrsl Could you clarify which CSS isn't being shown? If it's any style on the web pages HTML Publisher publishes, rather than the CSS of the wrapper frame, try [reading the fine manual](https://wiki.jenkins-ci.org/display/JENKINS/Configuring+Content+Security+Policy#ConfiguringContentSecurityPolicy-HTMLPublisherPlugin).

**Comment by GabrielleBreeze** [comment]
In Chrome browser Version 55.0.2883.95 (64-bit)
facing issues with html publisher plugin v.1.12; Jenkins LTS latest
`Refused to apply inline style because it violates the following Content Security Policy directive: "style-src 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-nACFrgFku/kPexwV/vzj0fjifssr4F2OK2ZayN8C00w='), or a nonce ('nonce-...') is required to enable inline execution.`

In FireFox browser Version 50.1.0 everything is fine.

**Comment by KaibutsuX** [comment]
Same issue here, command line args for -Dhudson.model.DreictoryBrowserSupport.CSP with and without fields have no effect

**Comment by rbywater** [comment]
@KaibutsuX once you have made the change, it's very hard to get Chrome to force reload the frame that holds the HTML. Therefore have you tried clearing your cache or (as the only way I've found) right click on the HTML content, select "View Frame Source" and do a CTRL+F5 forced refresh in that window, and then do a F5 refresh back in the normal HTML report view in Jenkins.

**Pull request merged**